### PR TITLE
feat(span-details): Add a function to format timestamps with leading zero - [TET-291]

### PR DESCRIPTION
**Pull request by priscilawebdev**
Implement the GitHub suggestion: https://github.com/getsentry/sentry/issues/37114

**Before:**
image

**After:**

image

### CODE REVIEW

File: static/app/components/events/interfaces/spans/utils.tsx [hunk 1]
```diff
@@ -913,3 +913,41 @@ const ERROR_LEVEL_WEIGHTS: Record<TraceError['level'], number> = {
   sample: 2,
   info: 1,
 };
+
+/**
+ * Formats start and end unix timestamps by inserting a leading zero if needed, so they can have the same length
+ */
+export function getFormattedTimeRangeWithLeadingZero(start: number, end: number) {
```
Comment (priscilawebdev) [inline static/app/components/events/interfaces/spans/utils.tsx:920]: I was not sure if this function should be here or somewhere in the common `utils` folder

**Review by matejminar — APPROVED** [review]
- Can we add at least one test case for `getFormattedTimeRangeWithLeadingZero`?
- The logic of `getFormattedTimeRangeWithLeadingZero` is a bit hard to read/understand, any chance it could be replaced with moments fractional second format?

None of the above are blockers, LGTM.

**Comment by priscilawebdev** [comment]
thank you @matejminar . I wanted to get some feedback before adding tests...will add in a follow-up

**Comment by priscilawebdev** [comment]
Thanks for the feedback @matejminar 

> Can we add at least one test case for getFormattedTimeRangeWithLeadingZero?
  Added tests in this follow-up PR https://github.com/getsentry/sentry/pull/38292

> The logic of getFormattedTimeRangeWithLeadingZero is a bit hard to read/understand, any chance it could be replaced with moments fractional second format?

I tried to do it, but I was losing precision by using `moment().format("...")` or `moment.unix()`, so I decided to leave as it is

**Comment by mwarkentin** [comment]
@priscilawebdev i think we want trailing 0 instead of leading 0 as leading 0 would actually change to value.

**Comment by priscilawebdev** [comment]
@mwarkentin you're right... trailing zeros after the decimal point do not affect the value of a number. thanks for the heads up. will update it

**Comment by priscilawebdev** [comment]
created this PR https://github.com/getsentry/sentry/pull/38507

**Pull request merged**